src/main/java/org/threeten/extra/scale/UtcInstant.java

```diff
@@ -467,6 +473,9 @@ public int hashCode() {
     @Override
     @ToString
     public String toString() {
+        if (stringValue != null) {
```

Comment:
This code does not confirm to the [racy single-check idiom,](http://javaagile.blogspot.com/2013/05/the-racy-single-check-idiom.html).

It would be better to split the toString generation into a separate method as the hotspot inliner prefers small hot methods to large methods where most of the code is cool.

Reply:
I fixed this. See the updated PR.